// 显示声明，以表示使用proto3的语法
syntax = "proto3";

// 函数
service Greeter {
    // 入参是 message HelloRequest 这个格式，出参是 message HelloRequest 这个格式
    // 通过不同函数的调用来传参
    // 用这个的好处在于，免除自己压缩dict为二进制、再自己发送数据的麻烦
    rpc SayHello (HelloRequest) returns (HelloReply) {
    }
    rpc SendPeople (PeopleRequest) returns (IsCorrect) {
    }
}

// client请求时使用的数据格式
message HelloRequest {
    string name = 1;
}

// server回复时的数据格式
message HelloReply {
    string message = 1;
}

// 多个参数，人员信息请求
message PeopleRequest {
    string name = 1;
    int32 age = 2;
}

// 返回信息
message IsCorrect {
    bool isRight = 1;
}